code 3.1&3.2

qiuting1995 opened the pull request:
I wrote some codes in part 3.1&3.2, realize the calculation of the lowest cost path.

qiuting1995 committed:
code 3.1&3.2

I wrote some codes in part 3.1&3.2, realize the calculation of the lowest cost path.

troussil commented on inteligent_scissors/fLiveWireGetCostFcn.m:
```diff
@@ -0,0 +1,21 @@
+% Calculat the cost function
+% The gradient strength cost Fg
+dImg = double(dImg);
+[dY, dX] = gradient(dImg);
```
please document what does the 'gradient' function

troussil commented on inteligent_scissors/fLiveWireGetCostFcn.m:
```diff
@@ -0,0 +1,21 @@
+dFg = 1 - dFg./max(dFg(:));
+
+% The zero-crossing cost Fz
+lFz = ~edge(dImg, 'zerocross');
```
please recall what does the 'edge' function in a comment

troussil commented on inteligent_scissors/fLiveWireGetCostFcn.m:
```diff
@@ -0,0 +1,21 @@
+% The zero-crossing cost Fz
+lFz = ~edge(dImg, 'zerocross');
+
+% The gradient direction Fd ??
```
TODO

troussil commented on inteligent_scissors/fLiveWireGetCostFcn.m:
```diff
@@ -0,0 +1,21 @@
+
+% The gradient direction Fd ??
+% The Sum:
+dF = dWz.*double(lFz)+ dWg.*dFg;
```
Do not forget that costs are defined at pixel p from its 8 neighbors (horizontal, vertical and diagonal). Fg is for instance scaled by 1 for diagonal neighbors and by 1/sqrt(2) for horizontal and vertical neighbors. See the paper (eq. 3 and text around).

troussil commented:
Nice. Please, modify the code according to my comments and think about testing your code.

qiuting1995 committed:
Update fLiveWireGetCostFcn.m

qiuting1995 committed:
add Fd to complete the local cost

last time I used Fz and Fg to calculate the l(q). this time I add Fd to calculate the l(p,q). and with l(p,q) I modify some code in other functions.

qiuting1995 committed:
add livewire function
